### DISCUSSION TITLE
Avoid duplication in spans

### DISCUSSION THREAD

**Discussion by wduminy** [tower-rs/tower-http#457] [Q&A] [answered]
### Summary

In my log I get the JSON shown below.
I notice that the `spans` property generally contains information already available in `span`.

I think my example is an exceptional case -- but I do see `spans` containing `span`. 

Is there a way to remove this duplication? 

```json
{
    "timestamp": "2024-01-11T10:25:19.621394Z",
    "level": "DEBUG",
    "fields": {
        "message": "finished processing request",
        "latency": "0 ms",
        "status": 200
    },
    "target": "tower_http::trace::on_response",
    "span": {
        "requestId": "60495ea7-df49-4acf-8a69-2cd27d994da9",
        "xrayTraceId": "...snipped...",
        "name": "Lambda runtime invoke"
    },
    "spans": [
        {
            "requestId": "60495ea7-df49-4acf-8a69-2cd27d994da9",
            "xrayTraceId": "...snipped...",
            "name": "Lambda runtime invoke"
        }
    ]
}
```

### tower-http version

0.5.0

**Comment by davidbarsky** [comment, marked as answer]
Try setting [Json::with_span_list](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/format/struct.Json.html#method.with_span_list) *xor* [Json::with_current_span](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/format/struct.Json.html#method.with_current_span) to false. I personally recommend setting the latter to false, as it'd keep emitting parent more than the just the current span.
